akka-projection-grpc-integration/src/test/scala/akka/projection/grpc/replication/ReplicationProtoEventIntegrationSpec.scala

```diff
@@ -0,0 +1,356 @@
+import akka.testkit.SocketUtil
+import com.google.protobuf.CodedInputStream
+import com.google.protobuf.CodedOutputStream
+import com.typesafe.config.Config
+import com.typesafe.config.ConfigFactory
+import org.scalatest.BeforeAndAfterAll
+import org.scalatest.wordspec.AnyWordSpecLike
+import org.slf4j.LoggerFactory
+import scalapb.GeneratedMessage
+import scalapb.GeneratedMessageCompanion
+
+import java.nio.ByteBuffer
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import scala.concurrent.duration.DurationInt
+
+object ReplicationProtoEventIntegrationSpec {
+
+  class ProtobufSerializer(val system: ExtendedActorSystem)
+      extends SerializerWithStringManifest
+      with BaseSerializer
+      with ByteBufferSerializer {
+
+    private val TagChangedManifest = akka.projection.grpc.test.TagChanged.javaDescriptor.getName
+    private val GreetingChangedManifest = akka.projection.grpc.test.GreetingChanged.javaDescriptor.getName
+
+    override def manifest(o: AnyRef): String = {
+      o match {
+        case msg: GeneratedMessage => msg.companion.javaDescriptor.getName
+        case _ =>
+          throw new IllegalArgumentException(s"Can't serialize object of type ${o.getClass} in [${getClass.getName}]")
+      }
+    }
+
+    override def toBinary(o: AnyRef): Array[Byte] = {
+      o match {
+        case msg: GeneratedMessage => msg.toByteArray
+        case _ =>
+          throw new IllegalArgumentException(s"Cannot serialize object of type [${o.getClass.getName}]")
+      }
+    }
+
+    override def toBinary(o: AnyRef, buf: ByteBuffer): Unit = {
+      o match {
+        case msg: GeneratedMessage =>
+          val output = CodedOutputStream.newInstance(buf)
+          msg.writeTo(output)
+          output.flush()
+        case _ =>
+          throw new IllegalArgumentException(s"Cannot serialize object of type [${o.getClass.getName}]")
+      }
+    }
+
+    override def fromBinary(bytes: Array[Byte], manifest: String): AnyRef = {
+      getCompanion(manifest).parseFrom(bytes).asInstanceOf[AnyRef]
+    }
+
+    override def fromBinary(buf: ByteBuffer, manifest: String): AnyRef = {
+      getCompanion(manifest).parseFrom(CodedInputStream.newInstance(buf)).asInstanceOf[AnyRef]
+    }
+
+    private def getCompanion(manifest: String): GeneratedMessageCompanion[_] = manifest match {
+      case GreetingChangedManifest => akka.projection.grpc.test.GreetingChanged.messageCompanion
+      case TagChangedManifest      => akka.projection.grpc.test.TagChanged.messageCompanion
+      case unknown                 => throw new IllegalArgumentException(s"Unknown manifest $unknown")
+    }
+
+  }
+
+  private def config(dc: ReplicaId): Config =
+    ConfigFactory.parseString(s"""
+       akka.actor.provider = cluster
+       akka.actor {
+         serializers {
+           my-replication-serializer = "akka.projection.grpc.replication.ReplicationProtoEventIntegrationSpec$$ProtobufSerializer"
```

Comment:
wouldn't it be more common to use the built in proto serializer in Akka (enabled by default), or what is the reason for defining your own?

Reply:
Mostly didn't think of it, changed to that one now.